src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSourceProviders/DiagnosticSourceManager.cs

```diff
@@ -45,11 +45,11 @@ public DiagnosticSourceManager([ImportMany] IEnumerable<IDiagnosticSourceProvide
             .ToImmutableDictionary(kvp => kvp.Name, kvp => kvp);
     }
 
-    public ImmutableArray<string> GetDocumentSourceProviderNames()
-        => _nameToDocumentProviderMap.Keys.ToImmutableArray();
+    public ImmutableArray<string> GetDocumentSourceProviderNames(ClientCapabilities clientCapabilities)
```

Comment:
Allows us to filter the diagnostic sources based on the client (e.g. no task diagnostics in vscode).